plugins/google-sheets/src/sheets.ts

```diff
@@ -379,7 +388,11 @@ export async function syncSheet({
 
     const unsyncedItemIds = new Set(await collection.getItemIds())
 
-    const sheet = fetchedSheet ?? (await fetchSheetWithClient(spreadsheetId, sheetTitle))
+    const spreadsheetInfo = await fetchSpreadsheetInfo(spreadsheetId)
+    const sheetTitleFromId = spreadsheetInfo.sheets.find((sheet) => sheetId === sheet.properties.sheetId)?.properties.title
+    const sheetIdFromTitle = spreadsheetInfo.sheets.find((sheet) => sheetTitle === sheet.properties.title)?.properties.sheetId
+
+    const sheet = fetchedSheet ?? (await fetchSheetWithClient(spreadsheetId, sheetTitleFromId ?? sheetTitle))
```

Comment:
It has to be fetched via title, right? Because their API doesn't accept id?

https://github.com/framer/plugins/blob/85026c400997d879e12dab97294a1af783db9720/plugins/google-sheets/src/sheets.ts#L147

Reply:
There may be another API to just get a whole sheets values not using a range, but I don't wanna introduce too much change this late